### ISSUE TITLE
BinanceTimeInForce missing GTD

### ISSUE THREAD

**Issue by doublier1** [nautechsystems/nautilus_trader#1226]
# Bug Report

### Expected Behavior

Nautilus should handle "GTD" instruction in the timeInForce field.

Needs addition of `BinanceTimeInForce.GTD` and conversion to `TimeInForce.GTD`

See documentation here https://binance-docs.github.io/apidocs/futures/en/#public-endpoints-info

```
Time in force (timeInForce):

GTC - Good Till Cancel
IOC - Immediate or Cancel
FOK - Fill or Kill
GTX - Good Till Crossing (Post Only)
GTD - Good Till Date
```

### Actual Behavior

On Connect I get: 

2023-09-05T23:10:05.674726056Z [ERR] A23R-001.DataClient-BINANCE: Error on _connect: ValidationError("Invalid enum value 'GTD' - at `$.symbols[0].timeInForce[4]`")

### Steps to Reproduce the Problem

1. Connect to Binance (Testnet only tested)

### Specifications

OS platform: Ubuntu 22.04
Python version: 3.11
nautilus_trader version: 1.177

**Comment by cjdsellers** [comment]
Hi @doublier1 

Thanks for the report here, seems this was a recent addition to the Binance API - it was probably announced ahead of time but it's difficult to keep up with all the changes sometimes. The `msgspec` library we use for parsing has quite a strict way of handling undefined enum members, which results in the error you posted above.

This has actually been fixed on the `develop` branch from this commit 35d79aba15d4466d03f1c4dc157922b37e3dd7f4.
We're anticipating a release at the end of the week to get this fix out there.

**Comment by cjdsellers** [comment]
The remapping back to the Nautilus `TimeInForce.GTD` does still need to be added though.

**Comment by jcrist** [comment]
> The `msgspec` library we use for parsing has quite a strict way of handling undefined enum members, which results in the error you posted above.

Is this a bad thing? Having an error on decoding here lets you catch an unknown symbol early, rather than later on when processing where it might cause a more serious (or harder to debug) issue.

If you don't want to validate that these values are in a known set of strings, you could annotate the field as `str` instead of an enum, dropping this validation entirely.

We also recently added support for `Enum._missing_` hooks (a [documented](https://docs.python.org/3/library/enum.html#enum.Enum._missing_) but rarely used feature of the standard-library's enum types). If you wanted, you could add an `UNKNOWN` value to the enum, and map all unknown strings to that value:

```python
In [1]: import msgspec

In [2]: import enum

In [3]: class Example(enum.Enum):
   ...:     APPLE = "apple"
   ...:     BANANA = "banana"
   ...:     UNKNOWN = "unknown"
   ...:     @classmethod
   ...:     def _missing_(cls, name):  # called to handle any missing values
   ...:         return cls.UNKNOWN
   ...: 

In [4]: msgspec.json.decode(b'"apple"', type=Example)
Out[4]: <Example.APPLE: 'apple'>

In [5]: msgspec.json.decode(b'"banana"', type=Example)
Out[5]: <Example.BANANA: 'banana'>

In [6]: msgspec.json.decode(b'"cherry"', type=Example)
Out[6]: <Example.UNKNOWN: 'unknown'>
```

**Comment by cjdsellers** [comment]
> > The `msgspec` library we use for parsing has quite a strict way of handling undefined enum members, which results in the error you posted above.
> 
> Is this a bad thing? Having an error on decoding here lets you catch an unknown symbol early, rather than later on when processing where it might cause a more serious (or harder to debug) issue.

I think this is preferable, I'd much rather experience a parsing error on the message than a downstream error in the business logic. Ideally, we're looking ahead in the Binance release notes and anticipating any changes such as additional or removed enum members for certain fields.

**Comment by cjdsellers** [comment]
Fixed on `develop` branch and in the latest 1.179.0 release.
